tortoise/fields/relational.py

```diff
@@ -396,9 +404,16 @@ def __init__(
             unique = kwargs.pop("create_unique_index")
         super().__init__(field_type, unique=unique, **kwargs)
         self.validate_model_name(model_name)
-        self.model_name: str = model_name
+        self.model_name = model_name
         self.related_name: str = related_name
-        self.forward_key: str = forward_key or f"{model_name.split('.')[1].lower()}_id"
+        if not forward_key:
+            if not isinstance(model_name, str):
+                forward_key = f"{model_name.__name__.lower()}_id"
+            elif model_name == "self":
+                forward_key = "self_id"  # TODO: compatible
```

Comment:
Did you mean to leave the TODO? It would be nice to expand on what exactly needs to be done or create an issue.

Reply:
Remove support for 'self' as it requires a symmetrical argument for m2m, which have to change too many codes. I think it is better to do it in another PR.